### DISCUSSION TITLE
Payment succeeded email notification not being sent

### DISCUSSION THREAD

**Discussion by beneberle** [solspace/craft-freeform#115] [Questions & Answers] [answered]
I am running Freeform 3.10.6 Pro. I have a Stripe Integration (currently in Test Mode) set up on a form. The payment type is one-time, not a subscription. I added an HTML-based template that should be sent when stripe reports that payment was successful. Everything is working as expected re: payment, but the payment success notifications are not going out as far as I can tell. 

I added the webhook to trigger on the 4 events named in docs:

- customer.subscription.created
- customer.subscription.deleted
- invoice.payment_failed
- invoice.payment_succeeded

These events don't ever seem to fire when freeform handles the single payment: I *do* however see `charge.succeeded` events in the log. 

Is this because of Stripe's Test Mode? What am I missing? Do I need to  add `charge.succeeded` to the events that should trigger the webhook? Is freeform set up to respond to that?

Any help would be appreciated.

**Comment by beneberle** [comment]
I've been doing some digging and found this in the PaymentWebhooksController.php:

       `//TODO: implement all notification service call as events?
        //TODO: update payment/subscription status accordingly
        switch ($event->type) {
            case Event::CUSTOMER_SUBSCRIPTION_CREATED:
                $submissionId = $this->getSubmissionIdFromStripeEvent($event, $integration);
                $this->getPaymentsNotificationService()->sendSubscriptionCreated($submissionId);

                break;

            case Event::CUSTOMER_SUBSCRIPTION_DELETED:
                $submissionId = $this->getSubmissionIdFromStripeEvent($event, $integration);
                $this->getPaymentsNotificationService()->sendSubscriptionEnded($submissionId);

                break;

            case Event::INVOICE_PAYMENT_SUCCEEDED:
                $subscriptionId = $event->data->object->lines->data[0]->subscription;
                $subscription = $integration->getSubscriptionDetails($subscriptionId);
                if (!$subscription) {
                    throw new HttpException(400, 'Could not send successful payment notification');
                }

                $submissionId = $subscription['metadata']['submission'];

                $this->getPaymentsNotificationService()->sendSubscriptionPaymentSucceeded($submissionId);

                break;

            case Event::INVOICE_PAYMENT_FAILED:
                $subscriptionId = $event->data->object->lines->data[0]->subscription;
                $subscription = $integration->getSubscriptionDetails($subscriptionId);
                if (!$subscription) {
                    throw new HttpException(400, 'Could not send failed payment notification');
                }

                $submissionId = $subscription['metadata']['submission'];

                $this->getPaymentsNotificationService()->sendSubscriptionPaymentFailed($submissionId);

                break;

            default:
        }`

This would suggest that Freeform is not listening to the `charge` events surrounding single payments. 

Is there any workaround or way within Freeform to handle sending an email on the `charge.succeeded` event? I am fairly new to Craft but am not afraid of custom module/plugin work if necessary. I am hoping to avoid writing my own implementation for this.

**Comment by kjmartens** [comment]
Sorry for the delay... we'll have a developer look into this and update Freeform accordingly. :)

**Comment by kjmartens** [comment, marked as answer]
Hi @beneberle. We've added a fix for this (unreleased yet). Would you be able to try it out?

You'll need to set your `composer.json` Freeform to this:

```
"solspace/craft-freeform": "dev-master as 3.10.10"
```

And then run `composer update` so that this fixed version gets applied.

Are you able to do that and let me know if it resolves the issue at all for you? :)

**Comment by kjmartens** [comment]
@beneberle This is also released in Freeform 3.10.11, so you can now also try downloading the latest version. :)
